workshops/base_views.py

```diff
@@ -206,3 +206,19 @@ def get_success_url(self):
             return next_url
         else:
             return default_url
+
+
+class PrepopulationSupportMixin:
+
+    def get_initial(self):
+        return self.request.GET.dict()
+
+    def get_form(self, *args, **kwargs):
+        """Disable fields that are pre-populated."""
+        form = super().get_form(*args, **kwargs)
+        for field in self.get_initial():
+            try:
+                form.fields[field].disabled = True
```

Comment:
>This doesn't make autocomplete fields disabled.

It works for me. Can you point out the form where it fails to disable a field?

 >What about leaving pre-populated fields enabled?

I like having them disabled so that I do not accidently change them.